metadata-ingestion/src/datahub/sdk/datajob.py

```diff
@@ -89,22 +81,20 @@ def __init__(
     ):
         urn = DataJobUrn.create_from_ids(
             job_id=name,
-            data_flow_urn=str(flow_urn),
+            data_flow_urn=str(flow.urn),
         )
         super().__init__(urn)
         self._set_extra_aspects(extra_aspects)
-
-        if platform is None:
-            platform = self.urn.get_data_flow_urn().orchestrator
-        self._set_platform_instance(platform, platform_instance)
+        self._set_platform_instance(flow.urn.orchestrator, flow.platform_instance)
+        self._set_browse_path_from_flow(flow)
```

Comment:
do we have tests that verify the behavior of the datajob browse path generation under different possible inputs

Reply:
added tests with (no container) -> flow -> datajob, and container -> container -> flow -> datajob